lit-mar/lambda_function.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import print_function
+
+import urllib
+import sys
+import os
+import json
+import subprocess
+
+class Context(object):
+    def __init__(self, logger, metrics_writter, request, response, system_info):
+        self.logger = logger
+        self.metrics_writter = metrics_writter
+        self.request = request
+        self.response = response
+        self.system_info = system_info
+
+
+class BaseResponse(object):
+    """BaseResponse if the object providing standard response attributes and operations"""
+    def __init__(self, response_body=""):
+        """Constructor, construct a response
+
+        Parameters
+        ----------
+        response_body: Optionally give the response a non-empty body
+        """
+        self.headers = []
+        self.status_code = 200
+        self.body = response_body
+
+    def get_dict(self):
+        """Get the response in dict form"""
+        return {"headers":dict(self.headers), "body":self.body, "statusCode":self.status_code}
+
+    def add_response_property(self, key, value):
+        """Add a pair of property to response
+
+        Parameters
+        ----------
+        key: string
+            Key of the property
+        value: string
+            Value of the property
+        """
+        self.headers.append((key, value))
+
+    def set_response_status(self, code, reason_string=None):
+        """Set the response status
+
+        Parameters
+        ----------
+        code: int
+            Status code of the response
+        reason_string: string
+            Optional non-default response reason string
+        """
+        self.status_code = code
+
+    def set_response_body(self, response_body):
+        """Set the response body
+
+        Parameters
+        ----------
+        response_body: object
+            The body of response
+        """
+        self.body = response_body
+
+
+def lambda_handler(event, context):
+    """Calculate the outputs specified by the bound symbol.
+
+    Parameters
+    ----------
+    event: dict, list, str, int, float, or NoneType 
+        AWS Lambda uses this parameter to pass in event data to the handler. 
+    context: LambdaContext
+        AWS Lambda uses this parameter to provide runtime information to your handler. 
+
+    Returns
+    -------
+    response_dict: Returns the response of lambda_handler in dict.
+        In the HTTP response to the invocation request, serialized into JSON.
+    """
+    # load config
+    with open('config.json', 'r') as file:
+        config = json.load(file)
+    model_url = config["url_mar"]
+    # download and unzip MAR
+    urllib.urlretrieve(model_url, "/tmp/mar.zip")
+    subprocess.call("unzip /tmp/mar.zip -d /tmp", shell=True)
```

Comment:
what if download failed, and the file does not exist? Is the error msg clear for users?